provider/src/main/java/com/cloudera/director/google/sql/GoogleCloudSQLInstanceTemplateConfigurationProperty.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2015 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.cloudera.director.google.sql;
+
+import static com.cloudera.director.spi.v1.database.DatabaseServerInstanceTemplate.DatabaseServerInstanceTemplateConfigurationPropertyToken.ADMIN_PASSWORD;
+import static com.cloudera.director.spi.v1.database.DatabaseServerInstanceTemplate.DatabaseServerInstanceTemplateConfigurationPropertyToken.ADMIN_USERNAME;
+import static com.cloudera.director.spi.v1.database.DatabaseServerInstanceTemplate.DatabaseServerInstanceTemplateConfigurationPropertyToken.TYPE;
+
+import com.cloudera.director.spi.v1.model.ConfigurationProperty;
+import com.cloudera.director.spi.v1.model.ConfigurationPropertyToken;
+import com.cloudera.director.spi.v1.model.util.SimpleConfigurationPropertyBuilder;
+
+
+/**
+ * Google Cloud SQL instance template configuration properties.
+ */
+public enum GoogleCloudSQLInstanceTemplateConfigurationProperty implements ConfigurationPropertyToken {
+
+  TIER(new SimpleConfigurationPropertyBuilder()
+      .configKey("tier")
+      .name("Tier")
+      .defaultValue("D1")
+      .defaultDescription("The tier of your database. This affects performance and how much you will be charged.")
+      .widget(ConfigurationProperty.Widget.OPENLIST)
+      .addValidValues(
```

Comment:
As far as I know this is a director-spi restriction. We do the same thing on compute side and it looks like there is no way to avoid it at this moment. Away from these hard-coded properties the  user still has the right to enter whatever he wants if it passes the validation which works with API.